clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4387,7 +4398,52 @@ public FenceProducersResult fenceProducers(Collection<String> transactionalIds,
 
     @Override
     public Uuid clientInstanceId(Duration timeout) {
-        throw new UnsupportedOperationException();
+        if (timeout.isNegative()) {
+            throw new IllegalArgumentException("The timeout cannot be negative.");
```

Comment:
This exception is not documented on the KIP. (Same for `IllegalStateException` below.)